src/scr_flush_sync.c

```diff
@@ -59,6 +61,10 @@ static int scr_flush_sync_data(scr_cache_index* cindex, int id, kvtree* file_lis
   }
   MPI_Barrier(scr_comm_world);
 
+  dataset_path_str = spath_strdup(dataset_path);
+  asprintf(&state_file, "%s/rank_%d.state_file", dataset_path_str, scr_my_rank_world);
+  scr_free(&dataset_path_str);
```

Comment:
Let's change this to use spath, even though it adds a line:
```suggestion
  spath* state_file_path = spath_dup(dataset_path);
  spath_append_strf(state_file_path, "rank_%d.state_file", scr_my_rank_world);
  char* state_file = spath_strdup(state_file_path);
  scr_path_delete(&state_file_path);
```